src/Behat/Behat/Output/Node/Printer/ListPrinter.php

```diff
@@ -83,9 +83,21 @@ public function printScenariosList(OutputPrinter $printer, $intro, $resultCode,
         $intro = $this->translator->trans($intro, [], 'output');
 
         $printer->writeln(sprintf('--- {+%s}%s{-%s}' . PHP_EOL, $style, $intro, $style));
-        foreach ($scenarioStats as $stat) {
+        foreach ($scenarioStats as $key => $stat) {
             $path = $this->relativizePaths((string) $stat);
-            $printer->writeln(sprintf('    {+%s}%s{-%s}', $style, $path, $style));
+
+            $stepStat = isset($stepStats[$key]) ? $stepStats[$key] : null;
+            $stepLine = $stepStat ? $this->extractLineNumber((string) $stepStat) : null;
+
+            if ($stepLine !== null) {
+                $lineNumber = $this->translator->trans('on_line_number', ['%line%' => $stepLine], 'output');
+                
+                $lineHelper = ' (' . $lineNumber . ')';
+            } else {
+                $lineHelper = '';
+            }
```

Comment:
I might structure this more like

```
$path = $this->relativizePaths((string) $stat);
$path = $this->appendFailingStepText($stat, $stepStats, $path);
$printer->writeln(sprintf('    {+%s}%s%s{-%s}', $style, $path, $lineHelper, $style));
```

Partly to reduce the diff in this method, and also I think then it probably wouldn't need as many new variables (the ->appendStepText function could just early `return $path` if there's nothing to append)?